plugin/src/main/java/org/opennms/integrations/pagerduty/PagerDutyForwarder.java

```diff
@@ -254,4 +274,22 @@ public static boolean testAlarmAgainstExpression(JexlExpression expression, Alar
         jc.set("alarm", alarm);
         return (boolean)expression.evaluate(jc);
     }
+
+    private class TaskConsumer implements Runnable {
+
+        @Override
+        public void run() {
+            while (true) { // TODO figure out the right way to do this loop -- .take() blocks until an element is ready
+                try {
+                    PagerDutyForwarderTask task = taskQueue.take();
+                    sendPDEvent(task.getReductionKey(), task.getPdEvent());
+                } catch (InterruptedException e) {
+                    // TODO need to handle InterruptedException here correctly
```

Comment:
Break the look and return - this will happen when the thread pool is being shut down.